### PR TITLE
Feature/zy/cla check

### PR DESCRIPTION
**Pull request by ssignik**
1.validate input parameter in review certificate
2.cla check

### CODE REVIEW

**Comment by shishupei** [comment]
开始门禁检查，等门禁检查通过后可以合入

File: openeulerSever/src/main/java/com/huawei/it/euler/ddd/infrastructure/cla/ClaService.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Data
+@Slf4j
```
Comment (LiYanghang00) [inline openeulerSever/src/main/java/com/huawei/it/euler/ddd/infrastructure/cla/ClaService.java:19]: 这个逐渐是否可以去掉

File: openeulerSever/src/main/java/com/huawei/it/euler/controller/UserController.java [hunk 2]
```diff
@@ -222,4 +227,22 @@ public JsonResponse<String> cancelCompatibilityAgreement(HttpServletRequest requ
         logUtils.insertAuditLog(request, uuid, "compatibility agreement", "cancel", "cancel compatibility agreement");
         return protocolService.cancelAgreement(ProtocolEnum.COMPATIBILITY_LIST_USAGE_STATEMENT.getProtocolType(), uuid);
     }
+
+    /**
+     * cla签署校验
+     *
+     * @param request request
+     * @return JsonResponse
```
Comment (LiYanghang00) [inline openeulerSever/src/main/java/com/huawei/it/euler/controller/UserController.java:235]: 修改成  是否签署cla

File: openeulerSever/src/main/java/com/huawei/it/euler/ddd/infrastructure/cla/ClaService.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    @Value("${cla.check}")
+    private String checkUrl;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    public boolean check(String email) {
+        String queryUrl = checkUrl + "?email=" + email;
+        HttpEntity<String> entity = new HttpEntity<>(null, null);
+        ResponseEntity<String> responseEntity = restTemplate.exchange(queryUrl, HttpMethod.GET, entity, String.class);
```
Comment (LiYanghang00) [inline openeulerSever/src/main/java/com/huawei/it/euler/ddd/infrastructure/cla/ClaService.java:32]: 如果失败是否需要加上重试

**Review by LiYanghang00 — APPROVED** [review]

**Comment by shishupei** [comment]
<table style="border-collapse: collapse" border=1>
    <tr>
        <th>检查项</th>
        <th>状态</th>
        <th>日志</th>
    </tr>

<tr>
    <td>Check代码检查</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/certification-server/5/5_Check代码检查.html">查看日志</a></td>
</tr>

<tr>
    <td>DT覆盖率</td>
    <td>7.31603%</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/certification-server/5/5_测试覆盖率.html">查看日志</a></td>
</tr>

<tr>
    <td>敏感信息扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/certification-server/5/5_敏感信息扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>漏洞扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/certification-server/5/5_漏洞扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>安全编码扫描</td>
    <td>&#9989;</td>
    <td><a href="https://opensourceways-ci.test.osinfra.cn/log/certification-server/5/5_安全编码扫描.html">查看日志</a></td>
</tr>

<tr>
    <td>TestPlan接口测试</td>
    <td>&#9989;</td>
    <td></td>
</tr>

<tr>
    <td>流水线链接</td>
    <td colspan="2"><a href="https://devcloud.cn-north-4.huaweicloud.com/cicd/project/3a76c1785dda4b13a399937d1978f240/pipeline/detail/128c1a12cee14cd3ae4059676f9c3c61/ef3e3e9e93fa450eb1414677e7d950c6">点击跳转</a></td>
</tr>
</table>

**Pull request merged**